build/l10n/l10n.js

```diff
@@ -15,33 +15,36 @@
 
   var L10n = navigator.mozL10n._getInternalAPI();
 
-  navigator.mozL10n.bootstrap = function(file, debug) {
-    var ctx = navigator.mozL10n.ctx = new L10n.Context(file);
-
-    if (debug) {
-      DEBUG = true;
-    }
+  navigator.mozL10n.bootstrap = function(file, config) {
+    // If LOCALE_BASEDIR is set, we're going to alert about missing strings
+    DEBUG =  config.LOCALE_BASEDIR !== '';
 
-    ctx.addEventListener('notfounderror', function(e) {
-      if (e.loc === 'en-US') {
-        throw e;
-      }
-    });
+    var ctx = navigator.mozL10n.ctx = new L10n.Context(file);
 
-    var error = addBuildMessage.bind(this, 'error');
-    var warn = addBuildMessage.bind(this, 'warn');
+    ctx.addEventListener('notfounderror', stopBuild);
+    ctx.addEventListener('duplicateerror', stopBuild);
 
     if (DEBUG) {
+      var error = addBuildMessage.bind(this, 'error');
+      var warn = addBuildMessage.bind(this, 'warn');
+
       ctx.addEventListener('fetcherror', error);
       ctx.addEventListener('manifesterror', warn);
       ctx.addEventListener('parseerror', warn);
       ctx.addEventListener('resolveerror', warn);
       ctx.addEventListener('notfounderror', error);
+      ctx.addEventListener('duplicateerror', error);
```

Comment:
so now in DEBUG mode you will have two eventlisteners on duplicateerror - one to stopBuild, another to error. Does it make sense?